tests/msc3083_test.go

```diff
@@ -93,6 +93,17 @@ func checkRestrictedRoom(t *testing.T, alice *client.CSAPI, bob *client.CSAPI, s
 	// Leaving the room works and the user is unable to re-join.
 	bob.LeaveRoom(t, room)
 	bob.LeaveRoom(t, space)
+
+	// Wait until Alice sees Bob leave the space. This ensures that Alice's HS
+	// has processed the leave before Bob tries rejoining.
```

Comment:
```suggestion
	// Wait until Alice sees Bob leave the space. This ensures that Alice's HS
	// has processed the leave before Bob tries rejoining, so that it rejects his
	// attempt to join the room.
```